src/coreclr/jit/emitarm64.h

```diff
@@ -576,6 +578,40 @@ static code_t insEncodeSveElemsize_dtype(instruction ins, emitAttr size, code_t
 // for the 'dtype' field.
 static code_t insEncodeSveElemsize_dtype_ld1w(instruction ins, insFormat fmt, emitAttr size, code_t code);
 
+// Encodes an immediate value in consecutive bits from most signficant position 'hi' to least significant
+// position 'lo'.
+template <const size_t hi, const size_t lo>
+static code_t insEncodeUimm(size_t imm)
+{
+    static_assert(hi >= lo && hi < sizeof(code_t) * 8);
+    const size_t imm_bits = hi - lo + 1;
+    const size_t imm_max  = 1 << imm_bits;
+    assert(imm < imm_max);
+    return static_cast<code_t>(imm << lo);
```

Comment:
Should we have an assert that we're not losing any high-order bits? E.g.,
```
code_t result = static_cast<code_t>(imm << lo);
assert((result >> lo) == imm);
return result;
```
?

Reply:
I think it's worth adding this as it can tighten any portability assumptions that come with `code_t` and `size_t`.